src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -1104,10 +1238,15 @@ internal static object GetModelStateValue(ViewContext viewContext, string key, T
         }
 
         protected virtual TagBuilder GenerateLink(
-            [NotNull] string linkText,
-            [NotNull] string url,
+            string linkText,
+            string url,
```

Comment:
Oooo definitely not good, we can't be throwing in `TagHelper`s like that. If we get incomplete data we'll typically use logging to notify the user that incomplete information was provided to the `TagHelper`. For example the `ScriptTagHelper` uses a mode matcher to [log details](https://github.com/aspnet/Mvc/blob/a68d9e4cb1328b4f25683ac7eab422c190cd6eb6/src/Microsoft.AspNet.Mvc.TagHelpers/Internal/ModeMatchResult.cs#L59-L68). `AnchorTagHelper` shouldn't do the same thing but it definitely shouldn't explode :smile: